source/slang/slang-emit-spirv.cpp

```diff
@@ -6117,6 +6098,14 @@ struct SPIRVEmitContext : public SourceEmitterBase, public SPIRVEmitSharedContex
                 getID(varInst));
         }
 
+        // HACK: Not sure how this should function?
```

Comment:
looks like we need to duplicate `isIntegralScalarOrCompositeType` and have a utility function named `isIntegralOrLogicalScalarOrCompositeType` that returns `true` for bool scalars and vectors. And then we make `needFlatDecorationForBuiltinVar` call `isIntegralOrLogicalScalarOrCompositeType` instead, so that `isFlat` will be true for the `CullPrimitive` builtin. That way we can get rid of this special case logic.